composer/profiler/profiler.py

```diff
@@ -450,9 +464,9 @@ def start(self) -> None:
             raise RuntimeError(
                 f"Attempted to start profiler event {self.name}; however, this marker is already started")
 
-        epoch = self.profiler.state.epoch
-        step = self.profiler.state.step
-        batch_idx = self.profiler.state.batch_idx
+        epoch = self.state.epoch
```

Comment:
Following up on the comment to change the ProfilerEventHandler to take a `Timestamp` object instead of `epoch` and `step`, can these lines be replaced with a call to `state.timer.get_timestamp()`?